### DISCUSSION TITLE
Long term plan for ruby-kafka

### DISCUSSION THREAD

**Discussion by sadhu89** [zendesk/ruby-kafka#964] [General] [answered]
Hello! What are the plans for this library in the medium/long term? After `racecar` migrated from `ruby-kafka` to `rdkafka-ruby`, I saw a few comments about slowing down the development of this library in favour of `rdkafka-ruby`.
See: 
- https://user-images.githubusercontent.com/1410645/101079509-d024b380-3564-11eb-9e1b-e4b9bcf39fcf.png
- https://github.com/zendesk/racecar/pull/97#issuecomment-525216249

We are trying to choose a library to write to Kafka, and saw those comments recommending `rdkafka-ruby` instead but it doesn't support IAM roles (A feature we need). So we are favouring `ruby-kafka` but we are a little bit concerned that the library could become unmaintained in the future.

**Comment by mensfeld** [comment, marked as answer]
As one of the contributors, I can say this library is pretty much abandoned and should no longer be considered for any production usage lasting long. As @dasch said in the comment and on a few other occasions, there is no point in trying to keep up with Kafka development when other languages use a well-established C binding and it gets more official support.

- racecar moved to librdkafka (https://github.com/zendesk/racecar/pull/97#issuecomment-525216249)
- waterdrop moved to librdkafka (https://github.com/karafka/waterdrop/pull/106)
- karafka moved to librdkafka (https://github.com/karafka/karafka/issues/431)
- phobos decided to stay (https://github.com/phobos/phobos/issues/132

IAM may not be supported but you can still use MSK via SASL + SSL. Here's the setup required to make it work with rdkafka: https://karafka.io/docs/Deployment/#karafka-configuration-for-aws-msk-sasl-ssl

On top of that, I do get reports of users of Karafka 1.4 (ruby-kafka version based) that highlight either design issues, stability issues, or bugs and I do not expect them to be fixed.

Overall: not recommended, I would consider it either in a maintenance mode or fully abandoned. 


Also, there is a KIP for this for librdkafka: https://cwiki.apache.org/confluence/display/kafka/kip-86%3a+configurable+sasl+callback+handlers I think it's just a matter of time and priorities.

**Comment by dasch** [comment]
Lots of folks are still using ruby-kafka, but there's no concerted effort to keep up with Kafka features; it's simply not feasible for such a complex protocol. I recommend using librdkafka based libraries and frameworks instead.

**Reply by mensfeld** [reply]
@dasch would you be ok with me creating a deprecation notice for the readme recommending moving to rdkafka or rdkafka backed tooling?

**Reply by dasch** [reply]
Sure
